plugins/repository-gcs/build.gradle

```diff
@@ -48,50 +49,78 @@ opensearchplugin {
+  implementation "com.google.cloud:google-cloud-storage:2.55.0"
+  implementation "com.google.cloud:google-cloud-core:2.59.0"
+  implementation "com.google.cloud:google-cloud-core-http:2.59.0"
+
+  runtimeOnly "com.google.guava:guava:33.4.0-jre"
+  runtimeOnly "com.google.guava:failureaccess:1.0.2"
+  compileOnly "com.google.errorprone:error_prone_annotations:2.38.0"
+
+  runtimeOnly "org.slf4j:slf4j-api:${versions.slf4j}" // 2.0.16 in bom
+  runtimeOnly "commons-codec:commons-codec:${versions.commonscodec}" // 1.18.0 in bom
+  implementation "com.google.api:api-common:2.52.0"
+  implementation "com.google.api:gax:2.69.0"
+  runtimeOnly "com.google.api:gax-httpjson:2.69.0"
+  implementation "org.threeten:threetenbp:1.7.0"
+  runtimeOnly "com.google.protobuf:protobuf-java-util:${versions.protobuf}"
+  runtimeOnly "com.google.protobuf:protobuf-java:${versions.protobuf}"
+  runtimeOnly "com.google.code.gson:gson:2.12.1"
+  runtimeOnly "com.google.api.grpc:proto-google-common-protos:2.60.0"
+  runtimeOnly "com.google.api.grpc:proto-google-iam-v1:1.55.0"
+  implementation "com.google.auth:google-auth-library-credentials:1.37.1"
+  implementation "com.google.auth:google-auth-library-oauth2-http:1.37.1"
+  runtimeOnly "com.google.oauth-client:google-oauth-client:1.39.0" // 1.39.0 in bom
+  implementation "com.google.api-client:google-api-client:2.7.2"
+  implementation "com.google.http-client:google-http-client:1.47.1"
+  runtimeOnly "com.google.http-client:google-http-client-gson:1.47.1"
+  runtimeOnly "com.google.http-client:google-http-client-appengine:1.47.1"
+  runtimeOnly "com.google.http-client:google-http-client-jackson2:1.47.1"
+  runtimeOnly "com.fasterxml.jackson.core:jackson-core:${versions.jackson}" // 2.18.2 in bom
+  runtimeOnly "io.opencensus:opencensus-api:0.31.1"
+  runtimeOnly "io.opencensus:opencensus-contrib-http-util:0.31.1"
+  implementation "com.google.apis:google-api-services-storage:v1-rev20250718-2.0.0"
+
+  implementation "org.checkerframework:checker-qual:3.49.0"
+
+  runtimeOnly "io.opentelemetry:opentelemetry-api:1.47.0"
+  runtimeOnly "io.opentelemetry:opentelemetry-context:1.47.0"
+  runtimeOnly "com.google.api.grpc:proto-google-cloud-storage-v2:2.55.0"
+  runtimeOnly "io.grpc:grpc-api:1.71.0"
 
-  api 'com.google.code.gson:gson:2.13.0'
-
-  runtimeOnly "com.google.guava:guava:${versions.guava}"
-  api 'com.google.guava:failureaccess:1.0.1'
-
-  api "com.google.http-client:google-http-client:${versions.google_http_client}"
-  api "com.google.http-client:google-http-client-appengine:${versions.google_http_client}"
-  api "com.google.http-client:google-http-client-gson:${versions.google_http_client}"
-  api "com.google.http-client:google-http-client-jackson2:${versions.google_http_client}"
+  testImplementation project(':test:fixtures:gcs-fixture')
+}
 
-  api 'com.google.oauth-client:google-oauth-client:1.34.1'
+compileJava {
+  configurations.runtimeClasspath.files.forEach {
+    if (it.name.contains("33.4.0-jre")) {
```

Comment:
Can we extract guava_version to a variable and reference it everywhere its used in this build.gradle file?

The rest of the PR looks good to me. Thank you @willyborankin !